pkgs/development/haskell-modules/make-package-set.nix

```diff
@@ -124,20 +124,17 @@ let
   haskellSrc2nix = { name, src, sha256 ? null, extraCabal2nixOptions ? "" }:
     let
       sha256Arg = if sha256 == null then "--sha256=" else ''--sha256="${sha256}"'';
-    in buildPackages.stdenv.mkDerivation {
-      name = "cabal2nix-${name}";
+    in buildPackages.runCommand "cabal2nix-${name}" {
```

Comment:
Wondering why `buildPackages` was used here in the first place…

Reply:
`git blame` reveals
 - https://github.com/NixOS/nixpkgs/pull/70192